pkg/monitor/datapath_trace.go

```diff
@@ -22,9 +22,9 @@ const (
 	// traceNotifyCommonLen is the minimum length required to determine the version of the TN event.
 	traceNotifyCommonLen = 16
 	// traceNotifyV0Len is the amount of packet data provided in a trace notification v0.
-	traceNotifyV0Len = 32
+	traceNotifyV0Len = 40
 	// traceNotifyV1Len is the amount of packet data provided in a trace notification v1.
-	traceNotifyV1Len = 48
+	traceNotifyV1Len = 56
```

Comment:
Can we do this @gandro?

Reply:
Oh good catch. Yeah, I believe this is a breaking change, we can't do this.

These structs are written to the perf events ring buffer by the datapath and read by user space. The events map is (to my knowledge) not replaced during upgrades (or downgrades for that matter), and so it will happen that the buffer will be full with events written with the shorter length, but userspace expects them to have the new (longer length) and thus will lead to invalid data being read.

I think we will have to introduce a version 2 and version 3 of the trace notify struct to achieve these modifications, and ensure the agent can deal with all 4 versions. Once we no longer need to support upgrading from Cilium v1.16, then we can get rid of v0 and v1.

Reply:
i'm kinda hung on this; it seems like an interface for this would best address the versioning issue, with the current struct working to provide the base notification information.

I'm going to experiment with that change.